evaluation/evaluation_metrics/evaluate_generation/generation_speed/utils/generator.py

```diff
@@ -1,51 +1,81 @@
 import csv
+import os
+import pandas as pd
+
 from tqdm import tqdm
 
+from app.config.settings import logger
+from app.processor.files_processor import ensure_data_directory_exists, generate_output_filename
+
+
+def generate_responses(params, data: pd.DataFrame, llm_client, model_prompt, batch_size=4):
+    DEFAULT_PARAMS = {
+        "n_predict": 400,
+        "temperature": 0.0,
+        "stop": ["<|end|>"],
+        "repeat_last_n": 64,
+        "repeat_penalty": 1.18,
+        "top_k": 40,
+        "top_p": 0.95,
+        "min_p": 0.05,
+        "tfs_z": 1.0,
+        "typical_p": 1.0,
+        "presence_penalty": 0.0,
+        "frequency_penalty": 0.0,
+        "mirostat": 0,
+        "mirostat_tau": 5.0,
+        "mirostat_eta": 0.1,
+        "stream": False,
+    }
+    DEFAULT_PARAMS["temperature"] = params["temperature"]
+    DEFAULT_PARAMS["n_predict"] = params["tokens_to_predict"]
```

Comment:
why not set these values directly in the DEFAULT_PARAMS?

Reply:
They are, just that in this case we are letting the user to have the control to change those values in case he/she wants